Feature/em 14 add model and migration other tables

Pongporopong opened the pull request.

Pongporopong committed:
chore: add Dockerfile setup for frontend

Pongporopong committed:
chore: install laravel/sanctum

Pongporopong committed:
feat: add model and migration for employee table

Pongporopong committed:
feat: add initial model and migration for other tables without foreign keys

applus-ian commented on backend/database/migrations/2025_04_11_025520_create_audit_logs_table.php:
```diff
@@ -0,0 +1,33 @@
+        Schema::create('audit_logs', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedBigInteger('user_id');
+            $table->enum('action', ['update', 'delete', 'create']); // to be reviewed!
```
use class enum

applus-ian reviewed (changes requested):
Use attribute casting to all models that use class enums.

Reference: https://laravel.com/docs/12.x/eloquent-mutators#attribute-casting

Pongporopong committed:
feat: Refactor for Enum Integration and Sanctum Token Expiry Setup

- Apply Enum Classes in Migrations & Models
- Add Gender/CivilStatus/Action Enums
- Configure Sanctum Token Expiration
